keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resolvers/ImageResolver.kt

```diff
@@ -79,44 +80,66 @@ class ImageResolver(
       deliveryConfig,
       artifact,
       environment.name
-    ) ?: throw NoImageSatisfiesConstraints(artifact.name, environment.name)
+    ) ?: throw NoArtifactVersionHasBeenApproved(artifact.name, environment.name)
```

Comment:
Throwing a more specific error here because I believe this propagates to the history view/users

Reply:
Nice!!! ❤️